lib/routes/account.js

```diff
@@ -323,35 +383,20 @@ module.exports = function (
             )
         }
 
-        function createResponse () {
-          if (request.query.keys !== 'true') {
-            return P.resolve({
-              sessionToken: sessionToken
-            })
+        function upsertDevice () {
+          if (! form.device) {
+            return P.resolve()
           }
 
-          var password = new Password(
-            authPW,
-            emailRecord.authSalt,
-            emailRecord.verifierVersion
-          )
-          return password.unwrap(emailRecord.wrapWrapKb)
-            .then(
-              function (wrapKb) {
-                return db.createKeyFetchToken({
-                  uid: emailRecord.uid,
-                  kA: emailRecord.kA,
-                  wrapKb: wrapKb,
-                  emailVerified: emailRecord.emailVerified
-                })
-              }
-            )
+          var operation = form.device.id ? 'updateDevice' : 'createDevice'
+
+          return db[operation](emailRecord.uid, sessionToken.tokenId, form.device)
             .then(
-              function (keyFetchToken) {
-                return {
-                  sessionToken: sessionToken,
-                  keyFetchToken: keyFetchToken
-                }
+              function (result) {
+                device = result
+              },
+              function (err) {
+                log.error({ op: 'account.login.device', err: err })
```

Comment:
Yep. My thinking is that `/account/create` and `/account/login` should not fail just because something went wrong with device registration. We'd still want to do the actual account operation. Is that wrong?